### PR TITLE
Added minimal extensibility to Web UI

### PR DESCRIPTION
**Pull request by nicolatimeus**
Adds support for injecting client side extensions to the Web UI from external bundles:

Extension can be used for the following purposes:

* Add custom tabs to **System** section of the sidenav
* Add custom tabs to the **Settings** section
* Implement new authentication methods

The communication between the main web ui and the extension is performed using JSNI glue code.
Extensions can be written in plain javascript or using GWT.

This PR also includes an example bundle that implements an extension using GWT.

### CODE REVIEW

**Comment by nicolatimeus** [comment]
Requires `net.alchim31.maven:yuicompressor-maven-plugin:1.5.1` maven plugin to be added to build and test dependencies

File: kura/examples/org.eclipse.kura.example.web.extension/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+Bundle-Vendor: Eclipse Kura
+Bundle-License: Eclipse Public License v1.0
+Bundle-RequiredExecutionEnvironment: JavaSE-1.8
+Bundle-ActivationPolicy: lazy
+Service-Component: OSGI-INF/*.xml,
+ OSGI-INF/example.xml
+Bundle-ClassPath: .
+Import-Package: javax.servlet;version="3.1.0",
+ javax.servlet.http;version="3.1.0",
+ org.eclipse.kura.web.api;version="1.0.0",
```
Comment (MMaiero) [inline kura/examples/org.eclipse.kura.example.web.extension/META-INF/MANIFEST.MF]: maybe better a range?

File: kura/org.eclipse.kura.web2.ext/pom.xml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	<artifactId>org.eclipse.kura.web2.ext</artifactId>
+	<version>1.0.0-SNAPSHOT</version>
+	<packaging>eclipse-plugin</packaging>
+
+	<properties>
+		<kura.basedir>${project.basedir}/..</kura.basedir>
+	</properties>
+
+	<build>
+		<resources>
```
Comment (MMaiero) [inline kura/org.eclipse.kura.web2.ext/pom.xml:32]: just a minor reflow needed here

File: kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/js/extensions.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+var ExtensionRegistry = function () {
```
Comment (MMaiero) [inline kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/js/extensions.js:11]: maybe we need a copyright header also here?

**Review by MMaiero — COMMENTED** [review]
Is this expected?
`2019-09-10T15:37:53,720 [qtp12271324-114] INFO  o.e.j.s.h.C.ROOT - org.eclipse.kura.example.web.extension.server.DummyAuthenticationServiceImpl: ERROR: The serialization policy file '/exampleext/E108368E6CB17314B26B06920B097A53.gwt.rpc' was not found; did you forget to include it in this deployment?
2019-09-10T15:37:53,722 [qtp12271324-114] INFO  o.e.j.s.h.C.ROOT - org.eclipse.kura.example.web.extension.server.DummyAuthenticationServiceImpl: WARNING: Failed to get the SerializationPolicy 'E108368E6CB17314B26B06920B097A53' for module 'https://10.200.12.14/exampleext/'; a legacy, 1.3.3 compatible, serialization policy will be used.  You may experience SerializationExceptions as a result.`

**Review by MMaiero — APPROVED** [review]

**Pull request merged**